kernel/platform.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2015-2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/kobject.h>
+#include <zephyr/syscall_handler.h>
+#if defined(CONFIG_PLATFORM_ABST_SUPPORT)
+#include <zephyr/sys/slist.h>
+#include <zephyr/irq.h>
+#include <zephyr/device.h>
+#include <zephyr/platform.h>
+#endif
+
+#define LOG_LEVEL CONFIG_PLATFORM_ABST_SUPPORT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(platform_abst);
+
+#if defined(CONFIG_PLATFORM_ABST_SUPPORT)
+__attribute__((weak)) int platform_specific_dev_config(const struct device *dev,
```

Comment:
why weaks here?

Reply:
This is required in case SOC/platform vendors need to implement their own config method. The function prototype might need further update as this PR is very initial draft version and need to align the architecture scope before going to detailed implementation.